virttest/libvirt_vm.py

```diff
@@ -2474,24 +2474,35 @@ def get_distro(self):
             session.close()
             return distro
 
-    def install_package(self, name):
+    def install_package(self, name, pkg_path=None, need_update=True):
         """
         Install a package on VM.
         ToDo: Support multiple package manager.
 
         :param name: Name of package to be installed
+        :param pkg_path: The full path of the package to be installed
+        :param need_update: True, do update if the package is already
+                            installed.
+                            False, do not update if the package is already
+                            installed.
         """
         session = self.wait_for_login()
         vm_distro = self.get_distro()
         try:
             # distro specific support for package manager
             if vm_distro.lower() == 'ubuntu':
                 query_cmd = "dpkg -l | grep %s" % name
-                cmd = "apt install -y %s" % name
+                if pkg_path:
+                    cmd = "apt install -y %s" % pkg_path
+                else:
+                    cmd = "apt install -y %s" % name
                 update_cmd = "apt upgrade -y %s" % name
             else:
                 query_cmd = "rpm -q %s" % name
-                cmd = "yum install -y %s" % name
+                if pkg_path:
+                    cmd = "yum install -y %s" % pkg_path
+                else:
+                    cmd = "yum install -y %s" % name
                 update_cmd = "yum update -y %s" % name
```

Comment:
Could you help to take this opportunity to update this whole piece, such as:
1. Set install/update comand by detecting the distribution, if not ubuntu or redhat series, raise not supported error
2. Set install param
3. Perform the unified command